openapi-spec/swagger.yaml

```diff
@@ -2320,6 +2320,830 @@ paths:
+            example: 'sort=name:desc,id:asc'
+        - name: name
+          in: query
+          description: The storage host name
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: string
+        - name: id
+          in: query
+          description: Database ID created for a storage host.
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: string
+        - name: description
+          in: query
+          description: The storage host description
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: string
+        - name: native_storage_host_id
+          in: query
+          description: Actual ID of the storage host in the storage backend.
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: string
+        - name: status
+          in: query
+          description: The storage host status
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: string
+            enum:
+              - normal
+              - offline
+              - abnormal
+              - unknown
+        - name: ip_address
+          in: query
+          description: Ip address of the storage host.
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: string
+        - name: os_type
+          in: query
+          description: Operating system of the storage host
+          required: false
+          style: form
+          explode: true
+          schema:
+            type: string
+            enum:
+              - windows
+              - linux
+      responses:
+        '200':
+          description: List storage hosts query was success
+          content:
+            application/json:
+              schema:
+                type: object
+                required:
+                  - storage_hosts
+                additionalProperties: true
+                properties:
+                  storage_hosts:
+                    type: array
+                    title: the storage hosts schema
+                    items:
+                      $ref: '#/components/schemas/StorageHostRespSpec'
+        '401':
+          description: NotAuthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorSpec'
+        '403':
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorSpec'
+        '500':
+          description: An unexpected error occured.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorSpec'
+  '/v1/storages/{storage_id}/storage-host-groups':
```

Comment:
I think there is no much information from these group related GET APIS. User only get the native_id. we don't have a GET for details of native_id also. I believe we should provide some more information in these APIs like number of hosts , hosts etc.. My suggestion is, let us enhance these group related APIS and add later. Currently masking-views, storage-hosts, and storage-initiators are enough for the user to experience the host mapping use-case we support now..

Reply:
Yes, groups expected contain some details about individual items in future. Removed group APIs currently from spec